src/middlewared/middlewared/plugins/smb_/registry_share.py

```diff
@@ -372,6 +373,9 @@ async def share_to_smbconf(self, conf_in, globalconf=None):
         else:
             conf["nt acl support"] = "no"
 
+        if data["cluster_volname"]:
+            conf["glusterfs: volume"] = data["cluster_volume"]
```

Comment:
This will raise a `KeyError`
```suggestion
            conf["glusterfs: volume"] = data["cluster_volname"]
```